packages/ra-core/src/routing/useGetPathForRecord.ts

```diff
@@ -52,33 +52,21 @@ export const useGetPathForRecord = <RecordType extends RaRecord = RaRecord>(
     const getPathForRecord = useGetPathForRecordCallback<RecordType>(options);
 
     // we initialize the path with the link value
-    const [path, setPath] = useState<string | false | undefined>(() => {
-        getPathForRecord({
-            record,
-            resource,
-            link,
-        }).then(resolvedLink => {
-            if (resolvedLink === false) {
-                // already set to false by default
-                return;
-            }
-            // update the path when the promise resolves
-            setPath(resolvedLink);
-        });
-
-        return false;
-    });
+    const [path, setPath] = useState<string | false | undefined>();
```

Comment:
The path was initialized with `false`, now it's `undefined`. If it's on purpose, can you explain why?